cmd/meroxa/turbine/utils.go

```diff
@@ -472,7 +472,7 @@ func uploadFile(ctx context.Context, logger log.Logger, filePath, url string) er
 		return err
 	}
 
-	if res.Body != nil {
+	if res != nil && res.Body != nil {
```

Comment:
This looks fine, but seems like we need to add test coverage for this function and/or `UploadSource()`. Thoughts on filing a KTLO issue for this?

Reply:
good call out. adding unit tests for this file was the remaining task for this ticket https://github.com/meroxa/turbine-project/issues/251

most of the functions in this file will be hard to test because of external calls. with dependency injection or something like that, we'll see what we can do.